### DISCUSSION TITLE
Show missing books of a series

### DISCUSSION THREAD

**Discussion by luevano** [gotson/komga#1640] [Ideas]
This would show on a per series basis if there are missing books (and possibly which ones), and possibly too on the general statistics (media analysis?) of the server. Possibly show a generic "book" slot with nothing other than "book X" as name?

This would simply check for "gaps" in the books's numbering:

1. Book 1
2. Book 2
3. `<missing book>`
4. Book 4
5. `<missing book>`
6. Book 6

Then it would report "missing 2 books" and possibly which ones: "missing 2 books: 3, 6".

There is a feature like this in Mihon/Tachiyomi, I don't personally have a screenshot to show this but I found this issue that contains some screenshots of how it looks like: https://github.com/mihonapp/mihon/issues/80#issuecomment-1895820109

**Comment by gotson** [comment]
Komga has no knowledge of a series book ordering. Some series have weird numbering, like negative numbers, non-numerical numbers etc. Komga also has not metadata source to know whether books are missing or not.

It should be better handled in a downloader like Mylar.

**Reply by luevano** [reply]
It would just be a simple check for natural ordering, without knowledge of metadata, just finding gaps on main numbering (as in the example, no care for X.Y numbering). Could be optional, as to not pollute with "extra" info that not all series would need.

**Reply by gotson** [reply]
it's not something i want to add in Komga at the moment
